packages/manager/apps/dedicated/client/app/ip/components/list/byoip/aggregate/ip-ip-byoip-aggregate.controller.js

```diff
@@ -0,0 +1,50 @@
+export default class IpByoipAggregateController {
+  /* @ngInject */
+  constructor($scope, IpByoipService) {
+    this.$scope = $scope;
+    this.IpByoipService = IpByoipService;
+    this.ip = $scope.currentActionData.ipBlock;
+    this.aggregationIps = [];
+    this.selectedAggregationIp = {};
+    this.isLoaded = false;
+    this.errorMessage = null;
+
+    IpByoipService.getAvailableAggregationConfigurations(this.ip.ipBlock)
+      .then((response) => {
+        this.aggregationIps = response.data;
+        if (this.aggregationIps.length) {
+          this.selectedAggregationIp = this.aggregationIps[0];
+        }
+      })
+      .catch((error) => {
+        this.errorMessage = error.data.message;
+      })
+      .finally(() => {
+        this.isLoaded = true;
+      });
+
+    this.$scope.cancelAggregate = () => this.cancelAggregate();
+    this.$scope.aggregate = () => this.aggregate();
```

Comment:
Avoid using scope, here there is no need for it.
You could directly use these funtions in the template prefixing them with `$ctrl.`